### PR TITLE
Advanced Asset Components

### PR DESCRIPTION
**Pull request by justinedelson**
This pull request adds three new components for use in the AEM Assets Metadata Editor to render XMP standard metadata
* History - renders the edit history (generally from the creative applications)
* Color Swatches - renders the color swatch information (generally in AI or INDD files)
* Fonts - renders the list of fonts used (again generally in AI and INDD files)

Due to limitations of the Metadata Schema editor, these can't be directly added to schemas. There are two supported methods:

1) Manual editing of the schema nodes (similar to what is described on http://adobe-consulting-services.github.io/acs-aem-commons/features/assets-geo-map.html)
2) textfields with the appropriate property name (`xmpMM:History` - history, `xmpTPg:Fonts` - fonts, `xmpTPg:Colorants` – color swatches) and marked as readonly (disabled) will be automatically replaced with the "advanced" component *if* the OSGi component "com.adobe.acs.commons.dam.impl.CustomComponentActivatorListServlet" is activated (and configured, although it has a reasonable default).

For the sake of completeness, the existing map component can be automatically activated using the property name `location` (the map component renders multiple properties, so it wouldn't make sense to map this to one specific property name).

The History component leverages Adobe product icons for Photoshop, Illustrator, InDesign, and Bridge. Permission to use these icons has been granted based on inclusion of the NOTICE file contained in this pull request. Similar notice information will also need to be posted as part of the documentation.

advanced-metadata-components-screenshot

### CODE REVIEW

**Comment by justinedelson** [comment]
I should also credit James Lockman and Ravi Kiran Papani for their help with this code.

**Comment by justinedelson** [comment]
For Illustrator files, the Color Swatches is a bit more complicated as Illustrator has the notion of Plates and Swatch Groups. The component automatically figures out how to properly render the data based on which properties/nodes are available.

advanced-metadata-components-screenshot2

File: content/src/main/content/jcr_root/apps/acs-commons/components/dam/custom-component-activator/clientlib/activator.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    var encodedItemPath = encodeURIComponent(getQueryParameter("item"));
+
+    $(document).on("foundation-contentloaded", function(e) {
+/*        activate("xmpMM:History", "/apps/acs-commons/dam/content/admin/history");
+        activate("xmpTPg:Fonts", "/apps/acs-commons/dam/content/admin/fonts");
+        activate("xmpTPg:Colorants", "/apps/acs-commons/dam/content/admin/color-swatches");
+        activate("location", "/apps/acs-commons/dam/content/admin/asset-location-map");*/
+        $.get("/bin/acs-commons/dam/custom-components.json", function(data) {
```
Comment (davidjgonzalez) [inline content/src/main/content/jcr_root/apps/acs-commons/components/dam/custom-component-activator/clientlib/activator.js]: servlet context support?

Comment (justinedelson) [inline content/src/main/content/jcr_root/apps/acs-commons/components/dam/custom-component-activator/clientlib/activator.js]: good catch. fixed

**Comment by davidjgonzalez** [comment]
@justinedelson WDYT about changing the grid lines color to something more subdued? Maybe ##9B9B9B (or whatever the color Inks and Swatches text is?)

The dark borders pull my eyes away from the colors.

**Comment by justinedelson** [comment]
I went with #969696

**Pull request merged**